tsdb/engine/tsm1/engine.go

```diff
@@ -228,7 +258,8 @@ func (e *DevEngine) Begin(writable bool) (tsdb.Tx, error) {
 
 func (e *DevEngine) WriteTo(w io.Writer) (n int64, err error) { panic("not implemented") }
 
-func (e *DevEngine) writeSnapshot() error {
+// WriteSnapshot will snapshot the cache and write a new TSM file with its contents, releasing the snapshot when done.
+func (e *DevEngine) WriteSnapshot(async bool) error {
```

Comment:
yes, `log.Fatal` would be better. But I think that system level errors from the file operations should exit the process instead of simply logging it and continuing to operate.